src/Symfony/Component/Cache/Adapter/RedisAdapter.php

```diff
@@ -139,24 +158,48 @@ protected function doFetch(array $ids)
      */
     protected function doHave($id)
     {
-        return $this->redis->exists($id);
+        return (bool) $this->redis->exists($id);
     }
 
     /**
      * {@inheritdoc}
      */
     protected function doClear($namespace)
     {
-        // As documented in Redis documentation (http://redis.io/commands/keys) using KEYS
-        // can hang your server when it is executed against large databases (millions of items).
-        // Whenever you hit this scale, it is advised to deploy one Redis database per cache pool
-        // instead of using namespaces, so that FLUSHDB is used instead.
-        $lua = "local keys=redis.call('KEYS',ARGV[1]..'*') for i=1,#keys,5000 do redis.call('DEL',unpack(keys,i,math.min(i+4999,#keys))) end";
-
-        if (!isset($namespace[0])) {
-            $this->redis->flushDb();
-        } else {
-            $this->redis->eval($lua, array($namespace), 0);
+        // When using a native Redis cluster, clearing the cache cannot work and always returns false.
+        // Clearing the cache should then be done by any other means (e.g. by restarting the cluster).
+
+        $hosts = array($this->redis);
+        $evalArgs = array(array($namespace), 0);
+
+        if ($this->redis instanceof \Predis\Client) {
+            $evalArgs = array(0, $namespace);
+
+            $connection = $this->redis->getConnection();
+            if ($connection instanceof PredisCluster) {
+                foreach ($connection as $c) {
+                    $hosts[] = new \Predis\Client($c);
```

Comment:
This will also keep the PredisCluster itself in the array of hosts. Is it expected ?